### ISSUE TITLE
JIT_SIMD_VectorCast test fails when both FeatureSIMD and TieredCompilation are disabled

### ISSUE THREAD

**Issue by tannergooding** [dotnet/coreclr#21490]
The `JIT_SIMD_VectorCast_r` test is currently failing when both `FeatureSIMD` and `TieredCompilation` are disabled. The assert returned is:
```
Assert failure(PID 13624 [0x00003538], Thread: 9764 [0x2624]): Assertion failed 'vnWx != NoVN' in 'System.Numerics.Vector`1[Byte][System.Byte]:op_Explicit(struct):struct' (IL size 13)

    File: e:\users\tagoo\repos\coreclr\src\jit\valuenum.cpp Line: 1320
    Image: E:\Users\tagoo\Repos\coreclr\bin\tests\Windows_NT.x64.Checked\Tests\Core_Root\corerun.exe
```

**Comment by tannergooding** [comment]
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/x64_checked_windows_nt_jitx86hwintrinsicnosimd/310/ shows the failure. Given the list of new commits tested in the run, this looks related to https://github.com/dotnet/coreclr/pull/21386.

CC. @briansull

**Comment by briansull** [comment]
PR #21512 will fix this issue

**Comment by briansull** [comment]
Fixed with #21512
